llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -149,6 +149,11 @@ bool llvm::isVectorIntrinsicWithScalarOpAtArg(Intrinsic::ID ID,
   if (TTI && Intrinsic::isTargetIntrinsic(ID))
     return TTI->isTargetIntrinsicWithScalarOpAtArg(ID, ScalarOpdIdx);
 
+  // Vector predication intrinsics only demand the the first lane the last
+  // operand (the EVL operand).
```

Comment:
```suggestion
  // Vector predication intrinsics only demand the first lane of the EVL operand.
```

There's no reference to the last argument here, as this uses `VPIntrinsic::getVectorLengthParamPos`, so better to first mention that this is for the EVL operand?